docs/en/sql-reference/functions/other-functions.md

```diff
@@ -914,6 +914,54 @@ WHERE diff != 1
 
 Same as for [runningDifference](./other-functions.md#other_functions-runningdifference), the difference is the value of the first row, returned the value of the first row, and each subsequent row returns the difference from the previous row.
 
+## nonNegativeDerivative(metric, timestamp[, INTERVAL]) {#other_functions-nonnegativederivative}
+
+Calculates the growth ratio of given metric values by given timestamps.
+Returns 0 for the first row and derivative for all the consecutive rows.
+If at some point derivative is less than zero, this value is ignored and the result is 0.
+
+!!! warning "Warning"
```

Comment:
Do we need more these functions? I though `rowNumber`, `neighbor` should be deprecated and all stateful functions implemented as window functions.

Reply:
AFAIK, window functions do support only seconds interval (and larger)
Here I want to use sub-seconds also

Reply:
Ok, will make a window function out of this